osu.Game/Graphics/Carousel/Carousel.cs

```diff
@@ -348,6 +348,23 @@ private void updateItemYPosition(CarouselItem item, ref CarouselItem? previousVi
 
         #region Input handling
 
+        protected override bool OnKeyDown(KeyDownEvent e)
+        {
+            switch (e.Key)
+            {
+                case Key.Enter:
+                case Key.KeypadEnter:
+                    // this is a special hard-coded case to allow activating item with modifier keys pressed
+                    // (e.g. press Ctrl+Enter to start beatmap with autoplay mod selected).
+                    // We can't rely on GlobalAction.Select for that as it only responds to pressing the key without any modifiers.
+                    if (currentKeyboardSelection.CarouselItem != null)
+                        Activate(currentKeyboardSelection.CarouselItem);
+                    return true;
+            }
+
+            return base.OnKeyDown(e);
+        }
+
```

Comment:
Not 100% sure about the placement of this here, in old song select this is placed in `PlaySongSelect` while global action handling is in `SongSelect`.

I think having this where it is right now makes most sense, it encapsulates input handling of select action in a single file rather than two files as per old song select.

Reply:
It's not great but fine for now. I'd consider adding a global key for this to expose it to the user better.